documentation/src/main/asciidoc/topics/proc_storing_custom_java_objects.adoc

```diff
@@ -0,0 +1,42 @@
+[id='storing_binary_objects-{context}']
+= Storing Java Objects in Binary Format
+Java Hot Rod clients can handle Java objects that represent entities and
+perform marshalling to serialize and deserialize objects into `byte[]` arrays.
+C{plusplus}, C#, and Javascript Hot Rod clients can also handle objects in the
+respective languages.
+
+If you store entries in the cache as marshalled Java objects, you should configure the cache with the MediaType of the marshalled storage.
+
+.Procedure
+
+. Specify the MediaType that matches your marshaller implementation.
++
+* Protostream marshaller: Configure the MediaType as `application/x-protostream`.
+* JBoss marshalling: Configure the MediaType as `application/x-jboss-marshalling`.
+* StringMarshaller: Configure the MediaType as `text/plain`.
```

Comment:
@gustavonalle should we list all the marshaller options here? what about Kyro and ProtoStuff?

Reply:
@ryanemerson perhaps I should ask you this question too. here we list protostream, jboss marshalling, and the string marshaller. should we include kyro and protostuff?

Reply:
I don't think so, as we don't have clear documentation on what would be required to get these working on the server-side. Tbh they haven't had any attention since they were implemented from us or users AFAIK. IMO I think we should deprecate them \cc @tristantarrant.

Reply:
There's also the vanilla java serialization: ```application/x-java-serialized-object```

Regarding Kryo and protostuff, there is no transcoder for them currently, so conversions won't work

Reply:
@ryanemerson and @gustavonalle should kryo and protostuff be in docs for community only?

Reply:
Yes